### DISCUSSION TITLE
Sort data in view

### DISCUSSION THREAD

**Discussion by Kaherdin** [luyadev/luya#2168] [Questions & Problems]
I need to find a solution to sort data from Luya

First of all, I've got extraValue "pricings" which find all pricings entries.

```
//PricingBlock.php (view)
<?php foreach($this->extraValue('pricings') as $key => $pricing): ?>

//PricingBlock.php
  public function extraVars()
    {
     $pricings = Pricing::find()
                ->where(['branch_id' => 1])
                ->orderBy(['amount' => SORT_ASC])->all();
...
 }
```

Then, inside this loop, I've got another loop :

```
//PricingBlock.php (view)
<?php foreach($pricing->periods as $index => $period): ?>
   <li class="red-text"><?= $period['frequency'].' '.$period['description']?></li>
<?php endforeach; ?>     
```  
And that's working just fine :

```
//PricingBlock.php (view)
<ul>                                                        
   <li class="red-text">12 Abonnement annuel</li>
   <li class="red-text">6 Abonnement 6 mois</li>
   <li class="red-text">24 Abo 24 mois</li>
</ul>
```
But know, I need to sort the period by frequency, so I try to create a sort function to sort my array:

```
//PricingBlock.php (view)
<?php 
function cmp($a, $b)
{
    return $a['frequency'] <=> $b['frequency'];
}
usort($pricing->periods, "cmp");     
foreach($pricing->periods as $index => $period): ?><?php 
function cmp($a, $b)
{
    return $a['frequency'] <=> $b['frequency'];
}
usort($pricing->periods, "cmp");     
foreach($pricing->periods as $index => $period): ?>
...
<?php endforeach; ?>
```

But it doesn't sort at all. How can I debug more? I tried to make a var_dump ($pricings) to get more info, but I didn't get any useful information.

**Comment by nadar** [comment]
HI @Kaherdin 

+ You can either join the relation data and sort afterwards, something like this `Pricing::joinWith([periods])->sort(...)' this will also reduce sql querys and is the the way you should go.
+ Or Maybe you can use ArrayHelper::multisort (https://www.yiiframework.com/doc/api/2.0/yii-helpers-basearrayhelper#multisort()-detail) but i never tested this.

I hope this helps.

**Reply by nadar** [reply]
ps: instead of using var_dump you should use YIi Frameworks `Yii::debug` and then use the yii2-debug toolbar to take a look at the data.
